src/olympia/files/models.py

```diff
@@ -253,8 +253,8 @@ def generate_filename(self, extension=None):
         parts.append(self.version.version)
 
         if self.version.compatible_apps:
-            apps = '+'.join([a.shortername for a in
-                             self.version.compatible_apps])
+            apps = '+'.join(sorted([a.shortername for a in
+                             self.version.compatible_apps]))
```

Comment:
You'll need to indent this line further to match up with the `a.shortername` above to pass the `flake8` test that is failing on TravisCI.